Lecture3/Task2/Task2L3.java

```diff
@@ -0,0 +1,103 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Task2L3 {
+    public static void main(String[] args) {
+
+        List<Student> allStudents = initStudents ();
+
+        System.out.println("\nList of students:");
+        printStudentsList(allStudents);
+
+        System.out.println("\nLets sort students by age ASC:");
+        sortListByAge(allStudents);
+        printStudentsList(allStudents);
+
+        System.out.println("\nLets search");
+        printStudentsList(searchListByLastname("1", allStudents));
+
+        System.out.println("\nAverage age is: " + getListAverageAge(allStudents));
+
+        System.out.println("\nLets create an allStudentsMap");
+        Map<Integer, Student> allStudentsMap = createAllStudentsMap(allStudents);
+        printStudentsMap(allStudentsMap);
+
+        System.out.println("\nLets create a filterStudentsMap");
+        Map<Integer, Student> filterStudentsMap = filterStudentsMap(allStudentsMap);
+        printStudentsMap(filterStudentsMap);
+    }
+
+    private static List<Student> initStudents () {
+        List<Student> list = new ArrayList<>();
+        for (int i = 0; i < 10; i++) {
+            Student student = new Student();
+            student.setId(i + 1);
+            student.setName("Name" + (i + 1));
+            student.setLastname(((int)(Math.random() * 2) + "Lastname" + (i + 1)));
+            student.setAge(16 + (int)(Math.random() * (100 - 16)));
+            list.add(student);
+        }
+        return list;
+    }
+
+    private static void printStudentsList(List<Student> list){
+        System.out.print("Id:\t\t");
```

Comment:
46-49 lines: why do we need them?

Reply:
Just wanted to print first string with the names of columns

Reply:
fixed